meta-resin-common/recipes-support/resin-vars/resin-vars/os-networkmanager

```diff
@@ -1,43 +1,104 @@
-	lscmd="$lscmd /etc/NetworkManager/conf.d"
-	extra_confs=$(eval "$lscmd")
-	for extra_conf in $extra_confs; do
-		echo "Removing extra conf: $extra_conf"
-		rm "/etc/NetworkManager/conf.d/$extra_conf"
-	done
+# This defines config.json to NetworkManager values
+value_map='true=yes false=no'
+
+# Returnes the value of a key in a map variable
+# The map needs to be a space separated set of entries. Each key/value pair separated by '='
+# Returns:
+# 	If key exists in map: the value in 'value' variable.
+# 	If key not found: returns the key in 'value' variable.
+getValueInMap() {
+        local key="$1"
+        local map="$2"
+	# shellcheck disable=SC2001
+        value="$(echo "$map" | sed -e "s/.*${key}=\([^ ]*\).*/\1/")"
+	if [ "$value" = "$map" ]; then
+		value="$key"
+	fi
+}
+
+# Very simple logging function
+log() {
+	local level=$1
+	local msg=$2
+	echo "$tool: <$level> $msg"
+	if [ "$level" = "error" ]; then
+		echo "$tool: <$level> Exit due to fatal error."
+		exit 1
+	fi
 }
 
-if [ ! -z "$keys" ]; then
-	echo "Found custom config fragments in $CONFIG_PATH"
-	while read -r key; do
-		echo "Found custom config fragment: $key"
-		lscmd="$lscmd -I \"$key.conf\""
-		filter="'.os.networkManager.\"$key\"'"
-		cmd="jq -r $filter $CONFIG_PATH"
-		conf=$(eval "$cmd")
-		if [ -f "/etc/NetworkManager/conf.d/$key.conf" ] && [ "$(cat "/etc/NetworkManager/conf.d/$key.conf")" = "$conf" ]; then
-			echo "/etc/NetworkManager/conf.d/$key.conf already exists"
-			continue
-		else
-			echo "New config fragment or modified fragment detected. Writing /etc/NetworkManager/conf.d/$key.conf"
-			echo "$conf" > "/etc/NetworkManager/conf.d/$key.conf"
-		fi
-	done <<< "$keys"
-
-	if [ ! -f /etc/NetworkManager/conf.d/README ]; then
-		echo "Files here are auto-generated by os-networkmanager. Extra files will be removed" >> /etc/NetworkManager/conf.d/README
+#
+# MAIN
+#
+
+if [ ! -f "$CONFIG_PATH" ]; then
+	# shellcheck disable=SC1091
+	if [ -f /usr/sbin/resin-vars ]; then
+		. /usr/sbin/resin-vars
+	else
+		log "error" "Couldn't find /usr/sbin/resin-vars."
 	fi
+	if [ ! -f "$CONFIG_PATH" ]; then
+	       log "error" "No config.json found."
+	fi
+fi
+log "info" "Using $CONFIG_PATH config.json."
+
+# shellcheck disable=SC2153
+if [ -n "$NM_CONF_FRAGMENT" ]; then
+        nm_conf_fragment="$NM_CONF_FRAGMENT"
+fi
+log "info" "Using NetworkManager configuration fragment file in $NM_CONF_FRAGMENT."
+
+nm_config="# This file is generated based on os.networkManager configuration in config.json."
+
+# [Device]
+section="Device"
+s=0
+if ! config_json_value="$(jq -r " .os.network.wifi.randomMacAddressScan" "$CONFIG_PATH")"; then
+	log "error" "Error parsing $CONFIG_PATH."
+	exit 1
+fi
+if [ "$config_json_value" != "null" ]; then
+	log "info" "Found randomMacAddressScan configuration."
+	getValueInMap "$config_json_value" "$value_map"
+	if [ "$s" -eq 0 ]; then
+		nm_config="$(printf "%s\n[%s]\nwifi.scan-rand-mac-address=%s" "$nm_config" "$section" "$value")"
```

Comment:
Is there some nice way for building the nm_config value with string concatenation, so that we avoid the slight code duplication here - writing `wifi.scan-rand-mac-address=%s` in two places?

Reply:
Pushed a change to avoid this duplication.